### PR TITLE
Make more brave component updater code usable by ios

### PR DESCRIPTION
**Pull request by bridiver**
Resolves 

This is to assist with https://github.com/brave/brave-core/pull/19079/files

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_component_updater/browser/brave_component_updater_delegate.h [hunk 1]
```diff
@@ -3,27 +3,42 @@
...
+}
+
 class PrefService;
 
 namespace brave {
 
 class BraveComponentUpdaterDelegate : public BraveComponent::Delegate {
  public:
-  BraveComponentUpdaterDelegate();
+  using RegiserCallback = base::RepeatingCallback<void(
```
Comment (bsclifton) [inline components/brave_component_updater/browser/brave_component_updater_delegate.h:31]:
```suggestion
  using RegisterCallback = base::RepeatingCallback<void(
```

File: components/brave_component_updater/browser/brave_component_updater_delegate.h [hunk 2]
```diff
@@ -3,27 +3,42 @@
...
+      component_updater::ComponentUpdateService* cus,
+      const std::string& name,
+      const std::string& base64_public_key,
+      base::OnceClosure registered_callback,
+      BraveComponent::ReadyCallback ready_callback)>;
+  BraveComponentUpdaterDelegate(
+      component_updater::ComponentUpdateService* updater,
+      RegiserCallback register_callback,
```
Comment (bsclifton) [inline components/brave_component_updater/browser/brave_component_updater_delegate.h:39]:
```suggestion
      RegisterCallback register_callback,
```

File: components/brave_component_updater/browser/brave_component_updater_delegate.h [hunk 3]
```diff
@@ -47,9 +62,13 @@ class BraveComponentUpdaterDelegate : public BraveComponent::Delegate {
   PrefService* local_state() override;
 
  private:
+  raw_ptr<component_updater::ComponentUpdateService> component_updater_;
+  RegiserCallback register_callback_;
```
Comment (bsclifton) [inline components/brave_component_updater/browser/brave_component_updater_delegate.h:66]:
```suggestion
  RegisterCallback register_callback_;
```

File: browser/component_updater/brave_component_installer.h [hunk 4]
```diff
@@ -6,15 +6,13 @@
...
 #define BRAVE_BROWSER_COMPONENT_UPDATER_BRAVE_COMPONENT_INSTALLER_H_
 
-#include <stdint.h>
-
-#include <memory>
 #include <string>
 #include <vector>
 
 #include "base/files/file_path.h"
 #include "base/functional/bind.h"
```
Comment (tmancey) [inline browser/component_updater/brave_component_installer.h:13]: Unrelated to these changes but this appears to be unused

File: browser/component_updater/brave_component_updater_configurator.cc [hunk 5]
```diff
@@ -18,7 +18,6 @@
 #include "brave/components/constants/brave_switches.h"
 #include "build/build_config.h"
 #include "chrome/browser/browser_process.h"
-#include "chrome/browser/net/system_network_context_manager.h"
```
Comment (tmancey) [inline browser/component_updater/brave_component_updater_configurator.cc:21]: This was needed for `SystemNetworkContextManager` due to `chrome/browser/browser_process.h` forward declaring `SystemNetworkContextManager`

**Review by tmancey — APPROVED** [review]
LGTM++

File: browser/component_updater/brave_component_updater_configurator.cc (hunk 5 above)
Comment (bridiver) [reply browser/component_updater/brave_component_updater_configurator.cc:21]: `SystemNetworkContextManager` is no longer used in this casee

Comment (tmancey) [reply browser/component_updater/brave_component_updater_configurator.cc:21]: Ah, strange as I was looking at `g_browser_process->system_network_context_manager()
                ->GetSharedURLLoaderFactory()` which has clearly being removed :-)

File: chromium_src/chrome/browser/component_updater/chrome_component_updater_configurator.cc [hunk 6]
```diff
@@ -15,7 +18,10 @@ namespace component_updater {
 scoped_refptr<update_client::Configurator>
 MakeChromeComponentUpdaterConfigurator(const base::CommandLine* cmdline,
                                        PrefService* pref_service) {
-  return base::MakeRefCounted<BraveConfigurator>(cmdline, pref_service);
+  return base::MakeRefCounted<BraveConfigurator>(
+      cmdline, pref_service,
+      g_browser_process->system_network_context_manager()
```
Comment (bridiver) [inline chromium_src/chrome/browser/component_updater/chrome_component_updater_configurator.cc:23]: this change was to break the circular dependency for `chrome/browser/net/system_network_context_manager.h`

**Review by emerick — APPROVED** [review]

**Review by AlexeyBarabash — APPROVED** [review]
++

File: browser/component_updater/brave_component_updater_configurator.cc [hunk 7]
```diff
@@ -48,10 +46,12 @@ namespace component_updater {
...
     const base::CommandLine* cmdline,
-    PrefService* pref_service)
+    PrefService* pref_service,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
     : configurator_impl_(ComponentUpdaterCommandLineConfigPolicy(cmdline),
-        false),
-      pref_service_(pref_service) {
+                         false),
+      pref_service_(pref_service),
+      url_loader_factory_(url_loader_factory) {
```
Comment (goodov) [inline browser/component_updater/brave_component_updater_configurator.cc:54]: `std::move`

File: components/brave_component_updater/browser/brave_component_updater_delegate.cc [hunk 8]
```diff
@@ -64,11 +66,11 @@ BraveComponentUpdaterDelegate::GetTaskRunner() {
 }
 
 const std::string BraveComponentUpdaterDelegate::locale() const {
```
Comment (goodov) [inline components/brave_component_updater/browser/brave_component_updater_delegate.cc:68]: `const std::string&`

File: components/brave_component_updater/browser/brave_component_updater_delegate.h [hunk 9]
```diff
@@ -47,9 +55,12 @@ class BraveComponentUpdaterDelegate : public BraveComponent::Delegate {
   PrefService* local_state() override;
 
  private:
+  raw_ptr<component_updater::ComponentUpdateService> component_updater_;
+  raw_ptr<PrefService> local_state_;
```
Comment (goodov) [inline components/brave_component_updater/browser/brave_component_updater_delegate.h:59]: consider:
1. `raw_ref` here
2. `raw_ref<T>::from_ptr` in constructor
3. `base::to_address` in `PrefService* local_state()` getter.

**Review by goodov — APPROVED** [review]

**Comment by rillian** [comment]
This PR is flagged for Rust code review, but doesn't seem to include any Rust code?

**Comment by bridiver** [comment]
> This PR is flagged for Rust code review, but doesn't seem to include any Rust code?

@rillian  it got added during a rebase issue and I forgot to remove

**Pull request merged**